test/unit/spec/appWidgets/input/selectInputSpec.js

```diff
@@ -1,179 +1,235 @@
+                    expect(this.container.childElementCount).toBe(0);
                 });
         });
 
-        it('Should update value via bus', () => {
-            // select one value then update it.
-            const widget = SelectInput.make(testConfig);
-            return widget.start({ node: container }).then(() => {
+        describe('bus updates', () => {
+            it('Should update value via bus', async function () {
+                // select one value then update it.
+                const widget = SelectInput.make(this.testConfig);
+                await widget.start({ node: this.container });
+                const inputElem = this.container.querySelector('select[data-element="input"]');
+                expect($(inputElem).val()).toEqual(this.testConfig.parameterSpec.data.defaultValue);
+
+                bus.emit('update', { value: 'banana' });
                 return new Promise((resolve) => {
-                    bus.on('validation', (message) => {
-                        expect(message.isValid).toBeTruthy();
+                    bus.on('set-value', (msg) => {
+                        expect(msg).toEqual('banana');
+                        expect($(inputElem).val()).toEqual('banana');
                         resolve();
                     });
-                    bus.emit('update', { value: 'banana' });
                 });
             });
-        });
 
-        it('Should reset to default via bus', () => {
-            const widget = SelectInput.make(testConfig);
-            return widget.start({ node: container }).then(() => {
+            it('Should reset to default via bus', async function () {
+                const widget = SelectInput.make(this.testConfig, { initialValue: 'carrot' });
+                await widget.start({ node: this.container });
+                const inputElem = this.container.querySelector('select[data-element="input"]');
+
+                bus.emit('reset-to-defaults');
                 return new Promise((resolve) => {
-                    bus.on('validation', (message) => {
-                        expect(message.isValid).toBeTruthy();
+                    bus.on('set-value', (msg) => {
+                        expect(msg).toEqual(this.testConfig.parameterSpec.data.defaultValue);
+                        expect($(inputElem).val()).toEqual(
+                            this.testConfig.parameterSpec.data.defaultValue
+                        );
                         resolve();
                     });
-                    bus.emit('reset-to-defaults');
                 });
             });
         });
 
-        it('Should respond to input change events with "changed"', () => {
-            const widget = SelectInput.make(testConfig);
-            return widget.start({ node: container }).then(() => {
+        describe('change events', () => {
+            it('Should respond to input change events with "changed" and "validation" messages, single select', async function () {
```

Comment:
combined this and the next test into a single test, and created the counterpart for a multi select.